corehq/apps/userreports/models.py

```diff
@@ -464,11 +473,31 @@ def get_report_count(self):
         """
         return ReportConfiguration.count_by_data_source(self.domain, self._id)
 
+    def get_mirrored_engine_ids(self, server_env):
+        for env in self.mirrored_engine_ids:
+            if env.server_environment == server_env:
+                return env.engine_ids
+        return []
+
+    def validate_db_config(self):
+        mirrored_engine_ids = self.get_mirrored_engine_ids(settings.SERVER_ENVIRONMENT)
+        if not mirrored_engine_ids:
+            return
+        if self.engine_id in mirrored_engine_ids:
+            raise BadSpecError("mirrored_engine_ids list should not contain engine_id")
+
+        for engine_id in mirrored_engine_ids:
+            if not connection_manager.engine_id_is_available(engine_id):
+                raise BadSpecError(
+                    "DB for engine_id {} is not availble".format(engine_id)
+                )
+
     def validate(self, required=True):
```

Comment:
@sravfeyn I thought you said this was only called on save

Reply:
Looks like it's not, it's getting called from the view  (I was guessing since it should)

Reply:
The save will never get called since this will be set on a static config. So, I will also add it in the pillow for extra safety.

Reply:
You could add it to `_get_datasource_config`